packages/shared/src/hooks/source/useSourceActionsFollow.ts

```diff
@@ -20,7 +20,7 @@ export function useSourceActionsFollow({
 }: UseSourceActionsFollowProps): UseSourceActionsFollow {
   const { displayToast } = useToastNotification();
 
-  const { feedSettings } = useFeedSettings();
+  const { feedSettings } = useFeedSettings({ enabled: false });
```

Comment:
This is tricky because you still use the result of the hook.

Now, the `enabled` means that this instance will never try to fetch the data and it relies the data is fetched already somewhere else (useQuery will return data it has even when not enabled).

Which means if you use this hook standalone when feedSettings was never fetched you will always get no result. 

I see the query has the stale time so it should be free after first fetch and until stale time expires so not sure what was the cause for disabling here?

Reply:
@capJavert Because it gets fetched on squad page, but it does not seem to be used by squad page, so in the cases where you enter directly into a squad it fetches unneeded data.

I disabled it in useSourceActionsFollow because the hook, by its name, and what it seemingly does, doesn't seem like it should be the one fetching the settings, so I assumed it was just an oversight.

Even when disabling the query in this particular hook, FeedPreferences still gets fetched on relevant pages like index and source by useFeedSettings, and the 'isFollowing' check which is the only thing this hook does with the data, seems to work just the same.

Reply:
Yeah, all valid points, but I have an issue with someone removing the hook that is fetching on squad page and now this hook no longer has data because its disabled. 

We should solve refetch with stale time so that when this hook mounts it just says "oh this data is valid still" and skips refetch. Thats how we usually do it on other places. Open to look into it quickly tomorrow together 🙌

Reply:
Sure! 

Could also be a good candidate for the persistent cache that I'm working on right now :)